core/plugins/org.fusesource.ide.camel.model.service.core/src/org/fusesource/ide/camel/model/service/core/util/PropertiesUtils.java

```diff
@@ -421,6 +421,10 @@ public static Object getTypedPropertyFromUri(AbstractCamelModelElement selectedE
 		if (CamelComponentUtils.isTextProperty(p)) {
 			return val;
 		}
+		
+		if (CamelComponentUtils.isClassProperty(p) && !p.getKind().equals(AbstractCamelModelElement.NODE_KIND_ELEMENT) ) {
```

Comment:
you are giving the argument to do the modification.... not to not doing it.

If it will change in next version, it will be best to avoid duplication to avoid to forget to remove it from one place.